src/formats/test.rkt

```diff
@@ -17,13 +18,24 @@
   (assert-program! stx)
   (assert-program-type! stx)
   (match-define (list 'FPCore (list args ...) props ... body) (syntax->datum stx))
+  (define arg-names (for/list ([arg args])
+                      (if (list? arg)
+                        (last arg)
+                        arg)))
```

Comment:
This code is pretty ugly and a little brittle. Don't we need to update the syntax checker?